projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -2909,24 +2926,43 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         }
         this.cdr.detectChanges();
         this.calculateGridSizes();
-        this.onGroupingDone.emit(this.sortingExpressions);
-
+        const newSortingExpressions: Array<ISortingExpression> = this.sortingExpressions;
+        if (JSON.stringify(oldSortingExpressions) !== JSON.stringify(newSortingExpressions)) {
```

Comment:
following on the comment regarding the interface this implementation won't do
you need a full diff similar to what Maya recommended in the original issue so that you can extract cols that had their state changed from grouped to ungrouped and vise versa.